packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1901,12 +1903,17 @@ export class ContainerRuntime
 			closeContainer: (error?: ICriticalContainerError) => this.closeFn(error),
 		});
 
-		this.scheduleManager = new ScheduleManager(
+		this.deltaScheduler = new DeltaScheduler(
+			this.innerDeltaManager,
+			createChildLogger({ logger: this.logger, namespace: "DeltaScheduler" }),
+		);
+
+		this.inboundBatchAggregator = new InboundBatchAggregator(
 			this.innerDeltaManager,
-			this,
 			() => this.clientId,
-			createChildLogger({ logger: this.logger, namespace: "ScheduleManager" }),
+			createChildLogger({ logger: this.logger, namespace: "InboundBatchAggregator" }),
 		);
+		this.inboundBatchAggregator.setupListeners();
```

Comment:
This is not needed because the logic inside it can be done in the constructor of `InboundBatchAggregator`. However, it doesn't have any public members so the alternative would be to not have a member in container runtime for it and do something like `void new InboundBatchAggregator ..." similar to how `ScheduleManagerCore` was used.

Reply:
I'd probably suggest moving the setupListeners content into the constructor as you mention, otherwise the constructor returns a kind-of-wrong object.

InboundBatchAggregator should really probably have a public `dispose()` that unregisters those listeners which would be the reason for retaining a reference rather than `void`ing it (to call `this.inboundBatchAggregator.dispose()` when the ContainerRuntime disposes).

Reply:
Good suggestion. I added dispose.